### PR TITLE
wait_until_available function along with options needs a bit of clari…

### PR DESCRIPTION
**Pull request by jordi-garcia**
#2572 

Issue Ref wait_until_available function along with options needs a bit of clarification #2572
https://github.com/puppetlabs/bolt/issues/2572

### CODE REVIEW

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 1]
```diff
@@ -2,16 +2,20 @@
...
 #
 # > **Note:** Not available in apply block
 Puppet::Functions.create_function(:wait_until_available) do
   # Wait until targets are available.
   # @param targets A pattern identifying zero or more targets. See {get_targets} for accepted patterns.
   # @param options A hash of additional options.
   # @option options [String] description A description for logging. (default: 'wait until available')
-  # @option options [Numeric] wait_time The time to wait. (default: 120)
-  # @option options [Numeric] retry_interval The interval to wait before retrying. (default: 1)
+  # @option options [Numeric] wait_time The time to wait. (default: 120 seconds)
```
Comment (lucywyman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:17]:
```suggestion
  # @option options [Numeric] wait_time The time to wait in seconds. (default: 120)
```

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
+  # @option options [Numeric] retry_interval The interval to wait before retrying. (default: 1 second)
```
Comment (lucywyman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:18]:
```suggestion
  # @option options [Numeric] retry_interval The interval to wait before retrying, in seconds. (default: 1)
```

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 3]
```diff
@@ -2,16 +2,20 @@
 
 require 'bolt/util'
 
-# Wait until all targets accept connections.
+# Wait until all targets accept connections. This function allows a plan execution to wait for a customisable
```
Comment (lucywyman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:5]:
```suggestion
# Wait until all targets accept connections. This function allows a plan execution to wait for a customizable
```

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 4]
```diff
@@ -2,16 +2,20 @@
 
 require 'bolt/util'
 
-# Wait until all targets accept connections.
+# Wait until all targets accept connections. This function allows a plan execution to wait for a customisable 
+# amount of time via wait_time option until a target connection can be restablished. The plan will proceed to
```
Comment (lucywyman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:6]:
```suggestion
# amount of time via wait_time option until a target connection can be reestablished. The plan will proceed to
```

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 5]
```diff
@@ -2,16 +2,20 @@
 
 require 'bolt/util'
 
-# Wait until all targets accept connections.
+# Wait until all targets accept connections. This function allows a plan execution to wait for a customisable 
+# amount of time via wait_time option until a target connection can be restablished. The plan will proceed to 
+# the next step if the connection fails to reconnect in the time specified (default: 120 seconds). A typical
+# use case for this function is the remote host being reboot as part of the plan execution.
+#
```
Comment (lucywyman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:9]:
```suggestion
#
```

**Comment by lucywyman** [comment]
A few changes that will fix typos and fix the Rubocop errors. Let me know if you'd rather I push directly to your branch if you don't have time to address these!

**Comment by jordi-garcia** [comment]
Do you mind pushing to my branch ?

Many thanks

Sent from my Galaxy



-------- Original message --------

**Comment by lucywyman** [comment]
Sweet - ready for docs review then merge. Thanks @jordi-garcia !

**Review by beechtom — APPROVED** [review]

File: bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb [hunk 6]
```diff
@@ -2,16 +2,20 @@
 
 require 'bolt/util'
 
-# Wait until all targets accept connections.
+# Wait until all targets accept connections. This function allows a plan execution to wait for a customizable
+# amount of time via wait_time option until a target connection can be reestablished. The plan will proceed to
+# the next step if the connection fails to reconnect in the time specified (default: 120 seconds). A typical
+# use case for this function is the remote host being reboot as part of the plan execution.
```
Comment (hestonhoffman) [inline bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb:8]:
```suggestion
# Wait until all targets accept connections. This function allows a plan execution to wait for a customizable
# amount of time via the `wait_time` option until a target connection can be reestablished. The plan proceeds
# to the next step if the connection fails to reconnect in the time specified (default: 120 seconds). A typical
# use case for this function is if your plan reboots a remote host and the plan needs to wait for the host to reconnect
# before it continues to the next step. 
```

Comment (lucywyman) [reply bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb]: Thanks Heston!

Comment (hestonhoffman) [reply bolt-modules/boltlib/lib/puppet/functions/wait_until_available.rb]: No probs!

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**